shell/platform/windows/BUILD.gn

```diff
@@ -134,6 +139,7 @@ source_set("flutter_windows_source") {
   deps = [
     ":flutter_windows_headers",
     "//flutter/fml:fml",
+    "//flutter/impeller/renderer/backend/gles",
```

Comment:
The Windows embedder uses Impeller's `ProcTableGLES` and `DescriptionGLES` to resolve OpenGLES functions and determine the GL version and extensions. This was the approach recommended by Impeller folks: https://discord.com/channels/608014603317936148/1173336353187041380/1182110838849536040

Reply:
why does impeller require the context to be current - do we combine creation of the proc table with looking up gl state? Could we split that functionality out of proc table so you could use it exclusively?

Reply:
> do we combine creation of the proc table with looking up gl state? 

Yes. The proc table uses the `DescriptionGLES` to determine capabilities and [examine extensions' functions](https://github.com/flutter/engine/blob/ae61286e4dec66562590a2db2a421bc38e6492d2/impeller/renderer/backend/gles/proc_table_gles.cc#L119-L129), and the `DescriptionGLES` constructor fails if the context isn't current.

> Could we split that functionality out of proc table so you could use it exclusively?

I think so. We could have a "raw" proc table that can be initialized without a GL context, does not have the description/capabilities, and includes potentially unsupported extension functions. The "safe" proc table would load the description/capabilities and remove unsupported functions. This might require a little experimentation to get the API right.